File: rentcar-flutter/lib/models/driver_license.dart
class DriverLicense {
  final String id;
  final String personId;
  final String driverName;
  final String licenseNumber;
  final String issueDate;
  final String expireDate;
  final String category;
  final int? experienceYears;
  final double rating;
  final int completedRentals;
  final bool valid;
  final String? createdAt;

  const DriverLicense({
    required this.id,
    required this.personId,
    required this.driverName,
    required this.licenseNumber,
    required this.issueDate,
    required this.expireDate,
    required this.category,
    this.experienceYears,
    required this.rating,
    required this.completedRentals,
    required this.valid,
    this.createdAt,
  });

  factory DriverLicense.fromJson(Map<String, dynamic> json) => DriverLicense(
        id: json['id'] as String,
        personId: json['personId'] as String,
        driverName: json['driverName'] as String,
        licenseNumber: json['licenseNumber'] as String,
        issueDate: json['issueDate'] as String,
        expireDate: json['expireDate'] as String,
        category: json['category'] as String,
        experienceYears: json['experienceYears'] as int?,
        rating: (json['rating'] as num).toDouble(),
        completedRentals: json['completedRentals'] as int,
        valid: json['valid'] as bool,
        createdAt: json['createdAt'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'personId': personId,
        'licenseNumber': licenseNumber,
        'issueDate': issueDate,
        'expireDate': expireDate,
        'category': category,
        if (experienceYears != null) 'experienceYears': experienceYears,
      };
}

File: src/main/java/ru/superu/rentcarregister/model/DriverLicense.java
package ru.superu.rentcarregister.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import ru.superu.rentcarregister.model.enums.LicenseCategory;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "driver_licenses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
@ToString
public class DriverLicense {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "person_id", unique = true, nullable = false)
    private Person person;

    @NotBlank
    @Size(max = 20)
    @Column(length = 20, unique = true, nullable = false)
    private String licenseNumber;

    @NotNull
    private LocalDate issueDate;

    @NotNull
    private LocalDate expireDate;

    @NotNull
    @Enumerated(EnumType.STRING)
    private LicenseCategory category;

    @Min(0)
    @Max(100)
    private Integer experienceYears;

    @DecimalMin("0.0")
    @DecimalMax("5.0")
    @Builder.Default
    @Column(precision = 3, scale = 2)
    private java.math.BigDecimal rating = java.math.BigDecimal.valueOf(5.0);

    @Builder.Default
    private Integer completedRentals = 0;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    private void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    public boolean isValid() {
        return expireDate != null && expireDate.isAfter(LocalDate.now());
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DriverLicense that = (DriverLicense) o;
        return id != null && id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
